### PR TITLE
detect/bsize: Validate against content buffer when available

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #5014

This PR adds additional validation when using the `bsize` keyword. If a one or more `content` keywords immediately precedes `bsize`, then the `bsize` value is checked against each to see if a match is possible using the operation (`=, <, >, <>`) and the value.

An error is raised if `bsize` value prevents a match, e.g., the content length exceeds the `bsize` value. The `bsize` operation and values are used to do the evaluation.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3682](https://redmine.openinfosecfoundation.org/issues/3682)

Describe changes:
- Address review comments

Companion [Suricata PR #233](https://github.com/OISF/suricata-verify/pull/233)

### CODE REVIEW

**Comment by catenacyber** [comment]
Could we use detect-engine-uint.c helper functions ?

**Comment by jlucovsky** [comment]
> Could we use detect-engine-uint.c helper functions ?

I don't think that would be a benefit in this case. `DetectCheckValue` has different semantics than the match function in `detect-engine-uint.c`.

File: doc/userguide/rules/payload-keywords.rst [hunk 1]
```diff
@@ -267,18 +267,43 @@ You can also use the negation (!) before isdataat.
...
+
+   alert dns any any -> any any (msg:"test bsize rule"; dns.query; content:"short"; bsize:<10; sid:124; rev:1;)
+
+.. container:: example-rule
+
+   alert dns any any -> any any (msg:"test bsize rule"; dns.query; content:"longer string"; bsize:>10; sid:125; rev:1;)
+
+.. container:: example-rule
+
+   alert dns any any -> any any (msg:"test bsize rule"; dns.query; content:"middle"; bsize:5<>15; sid:126; rev:1;)
```
Comment (inashivb) [inline doc/userguide/rules/payload-keywords.rst:305]: This should probably be `bsize:`6<>15` as length of "middle" is 6.

**Comment by jlucovsky** [comment]
Continued in #5576

File: doc/userguide/rules/payload-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/payload-keywords.rst:305]: What does `6<>15` mean ? A range ?

Comment (jlucovsky) [reply doc/userguide/rules/payload-keywords.rst:305]: Yes -- a range.

**Pull request closed without merging**